src/Workspaces/Core/Portable/Differencing/Match.cs

```diff
@@ -12,10 +12,9 @@ public sealed partial class Match<TNode>
     {
         private const double ExactMatchDistance = 0.0;
         private const double EpsilonDistance = 0.00001;
-        private const double MatchingDistance1 = 0.5;
-        private const double MatchingDistance2 = 1.0;
-        private const double MatchingDistance3 = 1.5;
-        private const double MaxDistance = 2.0;
+        private const double MatchingDistance1 = 0.25;
+        private const double MatchingDistance2 = 0.5;
+        private const double MaxDistance = 1.0;
```

Comment:
Shortly: This should be universally better than before. There should not be cases where we can get worse results.

We removed cases 1.5 and 2.0 which were never used properly with the current distance range of 
0...1. We added one more slice of 0.25.

Actually, the algorithm should be purely O(n^2). Instead of running the full algorithm, we try to make multiple passes hoping that they could be O(n). The more passes in the valid range, the better. We removed out of range passes and added a new one in the range.

Actually, I see that sometimes the algorithm is still O(n^2). Also we can try to invent a situation where adding more slices is required. Therefore, I would keep the issue open and think about further improvements later.